evm/gateway.js

```diff
@@ -199,17 +188,32 @@ async function processCommand(config, chain, options) {
         }
 
         case 'approveWithBatch': {
-            const { batchID, api } = options;
+            const { batchID } = options;
 
             if (!batchID) {
                 throw new Error('Batch ID is required for the approve action');
             }
 
             const batchId = batchID.startsWith('0x') ? batchID.substring(2) : batchID;
-            let apiUrl = api || `${config.axelar.lcd}/axelar/evm/v1beta1/batched_commands/${chain.name.toLowerCase()}`;
-            apiUrl = apiUrl.endsWith('/') ? apiUrl.slice(0, -1) : apiUrl;
+            const apiUrl = `${config.axelar.lcd}/axelar/evm/v1beta1/batched_commands/${chain.axelarId}/${batchId}`;
+
+            let executeData;
 
-            const executeData = await fetchBatchData(apiUrl, batchId);
+            try {
+                const response = await httpGet(`${apiUrl}`);
+
+                if (response == null || !response.execute_data) {
+                    throw new Error('Response does not contain execute_data');
+                }
+
+                if (response.status !== 'BATCHED_COMMANDS_STATUS_SIGNED') {
+                    throw new Error('Data is not yet signed by operators');
+                }
+
+                executeData = '0x' + response.execute_data;
+            } catch (error) {
+                throw new Error(`Failed to fetch batch data: ${error.message}`);
+            }
```

Comment:
the catch expression should only be on the fetch. currently it is catching other errors from if checks as well